src/components/results/sensitivityAnalysis/paged-sensitivity-analysis-result.js

```diff
@@ -0,0 +1,178 @@
+    FUNCTION_TYPES,
+    LOADER_TIMEOUT_MS,
+    PAGE_OPTIONS,
+} from './sensitivity-analysis-contants';
+import { useState, useRef, useEffect, useCallback, useMemo } from 'react';
+import { fetchSensitivityAnalysisResult } from '../../../utils/rest-api';
+import TablePaginationActions from '@mui/material/TablePagination/TablePaginationActions';
+import { useRowFilter } from '../../../hooks/use-row-filter';
+import { useIntl } from 'react-intl';
+import FilterPanel from '../../spreadsheet/filter-panel/filter-panel';
+import { SORT_WAYS, useAgGridSort } from '../../../hooks/use-aggrid-sort';
+
+const PagedSensitivityResult = ({
+    nOrNkIndex,
+    sensiKindIndex,
+    studyUuid,
+    nodeUuid,
+}) => {
+    const intl = useIntl();
+
+    const timerRef = useRef(null);
+
+    const [rowsPerPage, setRowsPerPage] = useState(DEFAULT_PAGE_COUNT);
+    const [page, setPage] = useState(0);
+    const [count, setCount] = useState(0);
+    const [result, setResult] = useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const [allowsShowingLoader, setAllowsShowingLoader] = useState(false);
+
+    const filtersDef = useMemo(
+        () => [
+            {
+                field: 'funcId',
+                label: intl.formatMessage({
+                    id: sensiKindIndex < 2 ? 'SupervisedBranches' : 'BusBarBus',
+                }),
+                options: result?.allFunctionIds || [],
+            },
+            {
+                field: 'varId',
+                label: intl.formatMessage({ id: 'VariablesToSimulate' }),
+                options: result?.allVariableIds || [],
+            },
+        ],
+        [intl, sensiKindIndex, result]
+    );
+
+    const { updateFilter, getFilterSelector } = useRowFilter(
+        DATA_KEY_TO_FILTER_KEY
+    );
+
+    // Add default sort on sensitivity col
+    const colKey = 'SENSITIVITY';
+    const sortWay = SORT_WAYS.desc;
+    const { onSortChanged, sortSelector } = useAgGridSort(
+        DATA_KEY_TO_SORT_KEY,
+        { colKey: colKey, sortWay }
+    );
+
+    const handleChangePage = useCallback((_, newPage) => {
+        setPage(newPage);
+    }, []);
+
+    const handleChangeRowsPerPage = useCallback((event) => {
+        setRowsPerPage(parseInt(event.target.value, 10));
+        setPage(0);
+    }, []);
+
+    const fetchResult = useCallback(async () => {
+        setIsLoading(true);
+
+        try {
+            const selector = {
+                isJustBefore: !nOrNkIndex,
+                functionType: FUNCTION_TYPES[sensiKindIndex],
+                offset: page * rowsPerPage,
+                chunkSize: rowsPerPage,
+                ...getFilterSelector(),
+                ...sortSelector,
+            };
+
+            setAllowsShowingLoader(false);
+            timerRef.current = setTimeout(() => {
+                setAllowsShowingLoader(true);
+            }, LOADER_TIMEOUT_MS);
+
+            const res = await fetchSensitivityAnalysisResult(
```

Comment:
we usually use .then()
remove async/await and use .then() to keep homogeneity()

Reply:
Done, thank you